### PR TITLE
run highlighted selections (partial chunks, multiple selections, etc.)

### LINKED ISSUE #1160: eval highlighted text in the editor
for #1152

### PR DESCRIPTION
**Pull request by hamilton**
Fixes #1160. This PR:

- [x] filters out md and css
- [x] allows multiple selections w/ CodeMirror's native stuff (`ctrl` and drag)
- [x] backs out the type from the first partially selected cell
- [x] automatically runs the entirety of a plugin cell even if it is partially highlighted
- [x] automatically runs the entirety of a highlighted _line_ from a fetch chunk (not the whole chunk, just the line)
- [x] test coverage for selection functionality

### CODE REVIEW

File: src/actions/actions.js [hunk 1]
```diff
@@ -194,35 +194,86 @@ function getChunkContainingLine(jsmdChunks, line) {
...
+    evalText: chunk.chunkContent,
+    evalType: chunk.chunkType,
+    evalFrags: chunk.evalFlags,
+  })
+}
+
 export function evaluateText(chunk = undefined) {
   return (dispatch, getState) => {
     const cm = window.ACTIVE_CODEMIRROR
     const doc = cm.getDoc()
```
Comment (bcolloran) [inline src/actions/actions.js]: can put `const { jsmdChunks } = getState()` here at the top of the function

File: src/actions/actions.js [hunk 2]
```diff
@@ -194,35 +194,86 @@ function getChunkContainingLine(jsmdChunks, line) {
...
     } else {
-      // FIXME handle case of code selected
-      // in this case, eval the first selection
-      // probably via codemirror doc.listSelections()[0],
-      // as long as the selection is contained by a code chunk
+      const selections = doc.getSelections()
+      const selectionLines = doc.listSelections()
+
+      const selectionSet = selections.map((sel, i) => {
+        // head & anchor depends on how you drag, so we have to sort by line+ch.
```
Comment (bcolloran) [inline src/actions/actions.js]: break `(sel, i) => ...` out into a function and add a few tests

**Comment by hamilton** [comment]
@bcolloran good points - will break out enough of this to reduce the size & consider the rest of your comment.

**Comment by hamilton** [comment]
@bcolloran this is ready for another review at your convenience.

**Pull request merged**